dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -1017,8 +1017,14 @@
 			<BTouchToolExe>bgen.dll</BTouchToolExe>
 			<BTouchToolPath>$(_XamarinSdkRootDirectory)\tools\lib\bgen</BTouchToolPath>
 			<BaseLibDllPath>$(_XamarinRefAssemblyPath)</BaseLibDllPath>
+			<BTouchCompiler>$(RoslynTargetsPath)/bincore/csc.dll</BTouchCompiler>
 			<_GeneratorAttributeAssembly>$(_XamarinSdkRootDirectory)/tools/lib/Xamarin.Apple.BindingAttributes.dll</_GeneratorAttributeAssembly>
 		</PropertyGroup>
+
+		<ItemGroup>
+			<_BTouchCompileCommand Include="$(DOTNET_HOST_PATH)" />
```

Comment:
When building from Windows these files will be the ones shipped in the SDK for Windows, and then copied to the Mac because of the ITaskItem[]. 

I'm not sure if that's going to work, shouldn't we reference this files from the right path on the Mac when building remotely instead?

Reply:
@emaf You mean move fetching the environment variable into the task itself?

Reply:
@emaf I implemented it a different way, please have a look again :)

Reply:
It does look better, but is this relative `$(RoslynTargetsPath)/bincore/csc.dll` or is it a full Windows path?

Reply:
@emaf I don't really know on Windows. but on macOS it's a full path, so I'd assume it at least can be on Windows as well.